IBAnimatable/TimingFunctionType.swift

```diff
@@ -0,0 +1,210 @@
+//
+//  Created by phimage on 19/04/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import UIKit
+
+public enum TimingFunctionType {
+
+  case linear
+  case easeIn
+  case easeOut
+  case easeInOut
+  case `default`
+
+  // #12 use a none or use optional with nil as default value in Animatable??
+  case none
+
+  // #12 sprint one?
+  // case spring(damping: CGFloat)
+
+  // #12 add a custom one?
+  // case custom(c1X Float, c1y: Float, c2x: Float, c2y: Float) // or named controlPoints
+
+  // http://easings.net/
+  case easeInSine, easeOutSine, easeInOutSine
+  case easeInCubic, easeOutCubic, easeInOutCubic
+  // #12 add more types
+
+}
+
+extension TimingFunctionType: IBEnum {
+
+  public init(string: String?) {
+    guard let string = string else {
+      self = .none
+      return
+    }
+
+    let (name, _) = TimingFunctionType.extractNameAndParams(from: string)
+
+    switch name {
+    // standards
+    case "linear":
+      self = .linear
+    case "easein":
+      self = .easeIn
+    case "easeout":
+      self = .easeOut
+    case "easeinout":
+      self = .easeInOut
+    case "default":
+      self = .default
+
+    // customs
+    case "easeInSine":
```

Comment:
It's a little bit confusing but `extractNameAndParams` returns only lowercased string (I will add it to the function documentation) so the `case` have to be lowercase to. (It's this way so case is ignored via InterfaceBuilder